test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -4334,6 +4476,274 @@ def test_cgroup_default_config(self):
         self.server.delete(id=jid, wait=True)
         self.assertFalse(self.is_dir(ehjd1, ehost1), "job cpuset dir found")
 
+    def test_cgroup_cgswap(self, vnode_per_numa_node=False):
+        if not self.mem:
+            self.skipTest('Test requires memory subystem mounted')
+        if not self.swapctl:
+            self.skipTest('Test requires memsw accounting enabled')
+        self.server.remove_resource('cgswap')
+        self.server.add_resource('cgswap', 'size', 'nh')
+        self.scheduler.add_resource('cgswap')
+        events = ['execjob_begin', 'execjob_launch', 'execjob_attach',
+                  'execjob_epilogue', 'execjob_end', 'exechost_startup',
+                  'exechost_periodic', 'execjob_resize', 'execjob_abort',
+                  'queuejob', 'modifyjob']
+        # Enable the cgroups hook new events
+        conf = {'enabled': 'True', 'freq': 10, 'event': events}
+        self.server.manager(MGR_CMD_SET, HOOK, conf, self.hook_name)
+
+        self.load_config(self.cfg15
+                         % ('true' if vnode_per_numa_node else 'false'))
+        # Restart mom for changes made by cgroups hook to take effect
+        begin = time.time()
+        # sleep 2s to ensure log matching will not catch older log lines
+        time.sleep(2)
+        self.mom.restart()
+        # Make sure the MoM is restarted
+        self.mom.log_match('Hook handler returned success'
+                           ' for exechost_startup',
+                           starttime=begin, existence=True,
+                           interval=1, tail=False)
+        self.server.expect(NODE, {'state': 'free'},
+                           id=self.mom.shortname, interval=2, offset=1)
+
+        vnode_name = self.mom.shortname
+        if vnode_per_numa_node:
+            vnode_name += "[0]"
+        cgswapstat = self.server.status(NODE, 'resources_available.cgswap',
+                                        id=vnode_name)
+        self.assertTrue(cgswapstat
+                        and 'resources_available.cgswap' in cgswapstat[0],
+                        'cgswap resource not found on node')
+
+        cgswap = PbsTypeSize(cgswapstat[0]['resources_available.cgswap'])
+        self.logger.info('Test node appears to have %s cgswap'
+                         % cgswap.encode())
+        if cgswap == PbsTypeSize("0kb"):
+            self.logger.info('First Mom has no swap, test will just '
+                             'check if job cgswap is added')
+            a = {'Resource_List.select':
+                 '1:ncpus=1:mem=100mb:vmem=1100mb:vnode=%s'
+                 % vnode_name}
+
+            j = Job(TEST_USER, attrs=a)
+            j.create_script(self.sleep30_job)
+            jid = self.server.submit(j)
+            # give the scheduler time to schedule, but job should not run
+            a = {'job_state': 'Q'}
+            self.server.expect(JOB, a, jid, offset=5)
```

Comment:
The scheduling cycle *is* already triggered by the job submission; there is no need to trigger it. The offset is 5 seconds in order to avoid consuming some attempts in the "expect" when we do not expect the job to run (it's been tuned to the behaviour of a well-known test harness). Some test harness virtual machines are fickle beasts that can take ages to do a scheduling cycle because of resource starvation (competition with other VMs on the host).

On my personal host, offset=1 is perfectly sane. It is futile to expect it to make tests faster on a well-known test harness with VMs.